app/qml/SearchBar.qml

```diff
@@ -105,23 +105,22 @@ Rectangle {
         anchors.centerIn: parent
         fillMode: Image.PreserveAspectFit
       }
+      
+      ColorOverlay {
+        anchors.fill: cancelSearchBtn
+        source: cancelSearchBtn
+        color: root.fontColor
+      }
 
       MouseArea {
-        x: cancelSearchBtn.x - cancelSearchBtn.width
-        height: cancelSearchBtn.height * 3
-        width: root.iconSize * 3
+        anchors.fill: iconContainer
+        z: 100
```

Comment:
It should be on the most top level indeed, but I think it is not necessary.

Reply:
i am paranoid

Reply:
@PeterPetrik unless we really need z=100, I would suggest to remove it. Explicit setting of z order can seriously mess up things (e.g. there is another panel on top of a panel with search bar, and this item may get precedence over some other item that is clearly on top of it...

Reply:
removed on master in https://github.com/lutraconsulting/input/commit/eff33712e245df402a5cb05135c82707b9ac577a